### PR TITLE
Minimum search query length

### LINKED ISSUE #1507: Investigate requiring a minimum length for text when searching content
Currently the search will display all results, which may not be what we want

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
- [x] Settings
  - [x] New settings are documented and present in `app.json`
  - [x] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
Closes #1507

#### What's this PR do?
- Sets a configurable minimum length for the search query parameter (default=2)
- Does not run the search if the query string is less than the minimum length.
- Shows an error message if the input is submitted and the query string is less than the minimum length.

#### How should this be manually tested?
- Under global and channel search, try submitting a search with 0, 1, and 2+ character search terms.
- You should see this with < 2-character searches:
screen shot 2019-01-10 at 10 44 19 am
- You should see results (or a `There are no results to display.` message depending on query) for 2+ character searches.

### CODE REVIEW

File: static/js/components/SearchTextbox.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
-// @flow
+// @flowarticle
```
Comment (noisecapella) [inline static/js/components/SearchTextbox.js]: Typo?

File: static/js/lib/validation.js [hunk 2]
```diff
@@ -185,6 +185,14 @@ export const validateContentReportForm = validate([
   )
 ])
 
+export const validateSearchQuery = (text: ?string): ?string =>
```
Comment (noisecapella) [inline static/js/lib/validation.js]: Can you add a test for this function?

File: static/js/containers/SearchPage_test.js [hunk 3]
```diff
@@ -107,6 +107,19 @@ describe("SearchPage", () => {
       )
     })
   })
+  ;["", "a"].forEach(query => {
+    it("doesn't run a search if initial search text is empty or < min characters", async () => {
```
Comment (noisecapella) [inline static/js/containers/SearchPage_test.js]: Can we alter the test name based on what `query` is?

File: static/js/components/SearchTextbox.js (hunk 1 above)
Comment (mbertrand) [reply static/js/components/SearchTextbox.js]: oops, yup

File: static/js/containers/SearchPage_test.js (hunk 3 above)
Comment (mbertrand) [reply static/js/containers/SearchPage_test.js]: ✔️

File: static/js/lib/validation.js (hunk 2 above)
Comment (mbertrand) [reply static/js/lib/validation.js]: ✔️

**Comment by noisecapella** [comment]
If I delete all text from the field all results disappear. I think we should preserve the results until the user clicks the search icon or presses enter

**Comment by mbertrand** [comment]
@noisecapella the previous search results should continue to display now until input submission.

**Pull request merged**